django/contrib/auth/management/commands/createsuperuser.py

```diff
@@ -185,6 +187,10 @@ def handle(self, *args, **options):
                         raise CommandError('You must use --%s with --noinput.' % field_name)
                     field = self.UserModel._meta.get_field(field_name)
                     user_data[field_name] = field.clean(value, None)
+                    if field.many_to_many and isinstance(user_data[field_name], str):
+                        user_data[field_name] = [pk.strip() for pk in (user_data[field_name]).split(",")]
+                    if field.many_to_one:
+                        user_data[field_name] = field.remote_field.model(field.clean(value, None))
```

Comment:
`test_fields_with_fk_by_env` and `test_fields_with_m2m_by_env` test this case. This is the actual solving of [#33151](https://code.djangoproject.com/ticket/33151).